pkg/secrets/passphrase/manager.go

```diff
@@ -111,15 +114,18 @@ func (sm *localSecretsManager) Encrypter() (config.Encrypter, error) {
 var lock sync.Mutex
 var cache map[string]secrets.Manager
 
+// clearCache is used to clear the cache, for tests.
+func clearCache() {
+	lock.Lock()
+	cache = nil
+	lock.Unlock()
+}
+
 func NewPassphaseSecretsManager(phrase string, state string) (secrets.Manager, error) {
 	// check the cache first, if we have already seen this state before, return a cached value.
 	lock.Lock()
```

Comment:
Nit: it would be nice to remove the open-coded locking by adding some accessor methods (e.g. `getSecretsManager(state string) (secrets.Manager, bool)` and `setSecretsManager(state string) (secrets.Manager, bool)`). I think there's also some TOCTOU bugs in here...